### ISSUE TITLE
Not able to parse int in my data class

### ISSUE THREAD

**Issue by ajay-dewari** [cbeust/klaxon#332]
In my data calss, I have a field quantity defined as below:

`val quantity : Int`

in the json it looks like:

`"quantity" : 3`

When I run the code I got the exception 

     Caused by: com.beust.klaxon.KlaxonException: Unable to instantiate Line:
     Parameter quantity: expected java.lang.Integer but received kotlin.Int (value: 2)
     
After this i undersstood the problem is with my data type, and the library by default cast the values into java data type, so I change it to  java.lang.Integer.

`val quantity : Integer`

even this dosn't helped and still I'm facing the same error with  `java.lang.Integer.`

     Caused by: com.beust.klaxon.KlaxonException: Unable to instantiate Line:
     Parameter quantity: expected java.lang.Integer but received java.lang.Integer (value: 2)

Since in the JSON the type of keys are not defined the library should cast based on the data class variable just like GSON and Jackson library do.

**Comment by joey-yoonsung** [comment]
I have the same issue. I got this error from every type of JSON field.

**Comment by arieagung** [comment]
You have to use null safe type, eg. Int?

**Comment by KeremAslan** [comment]
I have a similar issue:

```kotlin
Exception in thread "main" com.beust.klaxon.KlaxonException: Unable to instantiate GeoCoordinates:
    Parameter latitudeInDecDeg: expected kotlin.Float but received java.lang.Float (value: 70.0)

    Parameter longitudeInDecDeg: expected kotlin.Float but received java.lang.Float (value: 70.0)

    Parameter latLongResolutionInDp: expected kotlin.Int but received java.lang.Integer (value: 0)
```

This is my data class:
```kotlin
data class GeoCoordinates(val latitudeInDecDeg: Float, val longitudeInDecDeg: Float, val latLongResolutionInDp: Int = 4)
```

**Comment by jainsahab** [comment]
@cbeust Could you please help with this ? Why is Klaxon picking up the Java types as default. Are we all making a silly mistake here ?

**Comment by arieagung** [comment]
> I have a similar issue:
> 
> ```kotlin
> Exception in thread "main" com.beust.klaxon.KlaxonException: Unable to instantiate GeoCoordinates:
>     Parameter latitudeInDecDeg: expected kotlin.Float but received java.lang.Float (value: 70.0)
> 
>     Parameter longitudeInDecDeg: expected kotlin.Float but received java.lang.Float (value: 70.0)
> 
>     Parameter latLongResolutionInDp: expected kotlin.Int but received java.lang.Integer (value: 0)
> ```
> 
> This is my data class:
> 
> ```kotlin
> data class GeoCoordinates(val latitudeInDecDeg: Float, val longitudeInDecDeg: Float, val latLongResolutionInDp: Int = 4)
> ```

use null safe type: 
data class GeoCoordinates(val latitudeInDecDeg: Float?, val longitudeInDecDeg: Float?, val latLongResolutionInDp: Int? = 4)

**Comment by KeremAslan** [comment]
Ah, no it is not that. 

My data class has data validation in the `init` block which contained checks as `require()` or `check()`. The exception was being thrown by that `init` block, but what was happening is that Klaxon wraps that exception and throws its own exception without a stack trace. This makes it hard to debug.

**Comment by arieagung** [comment]
Yes, I also had that error, I just declare the variable with null-safe type then the error goes away.
